sm/download/src/download.rs

```diff
@@ -25,15 +34,25 @@ impl Downloader {
 
         let target_filename = PathBuf::new().join(target_dir_path).join(filename);
 
+        let config_location = TEdgeConfigLocation::from_default_system_location();
+        let config_repository = tedge_config::TEdgeConfigRepository::new(config_location.clone());
+        let tedge_config = config_repository.load().unwrap();
```

Comment:
This dependency must be removed. If a parameter is required, then add a function argument.

Reply:
I would simply not add a new configuration parameter. Retry and exponential backoff are hardcoded. The same can be done for this `reserved_buffer_size`.